packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -262,40 +265,135 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
          textHeightBehavior: textHeightBehavior,
          textWidthBasis: textWidthBasis,
        ),
-       _cursorColor = cursorColor,
-       _backgroundCursorColor = backgroundCursorColor,
        _showCursor = showCursor ?? ValueNotifier<bool>(false),
        _maxLines = maxLines,
        _minLines = minLines,
        _expands = expands,
-       _selectionColor = selectionColor,
        _selection = selection,
        _offset = offset,
        _cursorWidth = cursorWidth,
        _cursorHeight = cursorHeight,
-       _cursorRadius = cursorRadius,
        _paintCursorOnTop = paintCursorAboveText,
-       _cursorOffset = cursorOffset,
-       _floatingCursorAddedMargin = floatingCursorAddedMargin,
        _enableInteractiveSelection = enableInteractiveSelection,
        _devicePixelRatio = devicePixelRatio,
-       _selectionHeightStyle = selectionHeightStyle,
-       _selectionWidthStyle = selectionWidthStyle,
        _startHandleLayerLink = startHandleLayerLink,
        _endHandleLayerLink = endHandleLayerLink,
        _obscuringCharacter = obscuringCharacter,
        _obscureText = obscureText,
        _readOnly = readOnly,
        _forceLine = forceLine,
-       _promptRectRange = promptRectRange,
        _clipBehavior = clipBehavior {
     assert(_showCursor != null);
     assert(!_showCursor.value || cursorColor != null);
     this.hasFocus = hasFocus ?? false;
-    if (promptRectColor != null)
-      _promptRectPaint.color = promptRectColor;
+
+    _selectionPainter.highlightColor = selectionColor;
```

Comment:
For my understanding, the plan is to move all these painters out of this class?

Reply:
Ideally, all of them.